api/modules/files/controller.js

```diff
@@ -15,4 +18,32 @@ controller.data = function(req) {
   return data;
 };
 
+controller.delete = function(req, reply) {
+  var self = this;
+  var record = req.pre.records.models[0];
+
+  // If a published file exists for this file, we have
+  // to unset it's reference before deletion can occur
+  PublishedFiles.query({
+    where: {
+      file_id: record.get('id')
+    }
+  }).fetch()
+  .then(function(model) {
+    if (!model) {
+      return;
+    }
+    return model.set({
+      file_id: null
+    }).save();
+  })
+  .then(function(m) {
+    BaseController.prototype.delete.call(self, req, reply);
```

Comment:
You shouldn't have to do this because `BaseController`'s methods are set up in such a way that they user "self" **always**. And in the `delete` function, I don't see `this` being used anywhere